### PR TITLE
feat: initialize http client explicitly when calling go-oidc provider

### PR DESCRIPTION
**Pull request by takirala**
Addresses [D2IQ-75648](https://jira.d2iq.com/browse/D2IQ-75648)

`gatekeeper` can inject `HTTP_PROXY`, `HTTPS_PROXY` and `NO_PROXY` into tfa containers and we need to leverage these values when initializing an oidc client so that the communication happens via proxy when needed.

### CODE REVIEW

File: internal/configuration/config.go [hunk 1]
```diff
@@ -93,6 +94,10 @@ func NewConfig(args []string) (*Config, error) {
 
 	err := c.parseFlags(args)
 
+	// Set the client context explicitly in order to use proxy configuration from environment(if any)
+	// See https://github.com/coreos/go-oidc/blob/8d771559cf6e5111c9b9159810d0e4538e7cdc82/oidc.go#L43-L53
+	c.OIDCContext = oidc.ClientContext(context.Background(), &http.Client{})
```
Comment (jimmidyson) [inline internal/configuration/config.go:99]: Doesn't it use `http.DefaultClient` by default? Why doesn't that work with proxy env vars?

Comment (takirala) [reply internal/configuration/config.go:99]: acc. to https://github.com/coreos/go-oidc/issues/191 my understanding is http.DefaultClient uses http.DefaultTransport which does not initialize using the env here ?! Am i missing sth ?

Comment (jimmidyson) [reply internal/configuration/config.go:99]: Interesting. They must somehow set a default oauth2.Client that doesn't respect the environment... weird. Anyway, looks like you're right and this is needed.

Comment (jimmidyson) [reply internal/configuration/config.go:99]: Perhaps this is more related to golang.org/x/oauth2 in package than go-oidc but as it says in the go oidc docs, setting the client as you are doing here should work properly for both packages.

**Pull request merged**